google/resource_compute_network_endpoint_group.go

```diff
@@ -80,8 +80,8 @@ you create the resource.`,
 				Type:         schema.TypeString,
 				Optional:     true,
 				ForceNew:     true,
-				ValidateFunc: validation.StringInSlice([]string{"GCE_VM_IP_PORT", ""}, false),
-				Description:  `Type of network endpoints in this network endpoint group. Default value: "GCE_VM_IP_PORT" Possible values: ["GCE_VM_IP_PORT"]`,
+				ValidateFunc: validation.StringInSlice([]string{"GCE_VM_IP", "GCE_VM_IP_PORT", ""}, false),
```

Comment:
From the [API docs](https://cloud.google.com/compute/docs/reference/rest/beta/networkEndpointGroups) it looks like this has a number of potential values: GCE_VM_IP_PORT, NON_GCP_PRIVATE_IP_PORT, INTERNET_FQDN_PORT, INTERNET_IP_PORT, SERVERLESS, PRIVATE_SERVICE_CONNECT (but `GCE_VM_IP` doesn't seem to be an option)

But regardless - usually in cases like this (where the set of potential values changes frequently) we like to remove the validation on the Terraform side, so that we're not preventing users from using all available values that the API provides.

Reply:
it's little bit different with https://cloud.google.com/load-balancing/docs/negs

Reply:
Actually, looking into this some more, I think this validation is important - load balancers are weird.

Reply:
the another thing maybe the API currently doesn't support `GCE_VM_IP` type for network endpoint resource, but I agree to validate the type if the API doesn't support the API it should be return the API error. Just to avoid the network endpoint resource created with zero port. 

Based on experience if the network endpoint created with zero port it can't remove via IAC and we should remove manually from console and continue with remove network endpoint from state.

Reply:
Sorry, I was wrong about removing the validation. You should keep the change validation and add GCE_VM_IP to the list. Load balancers are unusual so we can't remove this the way we would with other resources.